### PR TITLE
ACS-5624 Search and Favorites API enhancements - E2E test

### PR DESCRIPTION
**Pull request by mpichura**

### CODE REVIEW

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java [hunk 1]
```diff
@@ -555,14 +561,28 @@ public void checkResponseSchemaForGetFavorites() throws Exception
...
+        //request allowable operations for favorites
+        final RestPersonFavoritesModelsCollection adminFavorites =
                 restClient.authenticateUser(adminUserModel).withCoreAPI().usingAuthUser().include(ALLOWABLE_OPERATIONS).getFavorites();
         restClient.assertStatusCodeIs(HttpStatus.OK);
 
         adminFavorites.getEntries().stream()
                 .map(RestPersonFavoritesModel::onModel)
                 .forEach(m -> m.assertThat().field(ALLOWABLE_OPERATIONS).isNotEmpty());
+
+        //Check if isFavorite field is returned when requested in search API.
```
Comment (tpage-alfresco) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java:574]: Why not have a second test for this?

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java [hunk 2]
```diff
@@ -555,14 +561,28 @@ public void checkResponseSchemaForGetFavorites() throws Exception
 
     @Test(groups = { TestGroup.REST_API, TestGroup.FAVORITES, TestGroup.REGRESSION })
     @TestRail(section = { TestGroup.REST_API, TestGroup.FAVORITES }, executionType = ExecutionType.REGRESSION, description = "Verify if get favorites response returns allowableOperations object when requested")
-    public void checkResponseForGetFavoritesWithAllowableOperations()
-    {
-        RestPersonFavoritesModelsCollection adminFavorites =
+    public void checkResponsesForGetFavoritesWithAllowableOperationsAndSearchRequestWithIsFavorite() throws InterruptedException {
```
Comment (tpage-alfresco) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java:564]: Trailing brace formatting. We should get around to investigating pre-commit formatters.

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java (hunk 1 above)
Comment (mpichura) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java:574]: Good point - will do.

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java [hunk 3]
```diff
@@ -553,16 +559,36 @@ public void checkResponseSchemaForGetFavorites() throws Exception
                 .field("title").is(thirdSiteModel.getTitle());
     }
 
-    @Test(groups = { TestGroup.REST_API, TestGroup.FAVORITES, TestGroup.REGRESSION })
-    @TestRail(section = { TestGroup.REST_API, TestGroup.FAVORITES }, executionType = ExecutionType.REGRESSION, description = "Verify if get favorites response returns allowableOperations object when requested")
-    public void checkResponseForGetFavoritesWithAllowableOperations()
+    @Test(groups = {TestGroup.REST_API, TestGroup.FAVORITES, TestGroup.REGRESSION})
+    @TestRail(section = {TestGroup.REST_API, TestGroup.FAVORITES}, executionType = ExecutionType.REGRESSION,
+            description = "Verify if get favorites response returns allowableOperations object when requested")
+    public void checkResponsesForGetFavoritesWithAllowableOperationsAndSearchRequestWithIsFavorite()
```
Comment (tpage-alfresco) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java:565]:
```suggestion
    public void checkResponsesForGetFavoritesWithAllowableOperations()
```

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java [hunk 4]
```diff
@@ -553,16 +559,36 @@ public void checkResponseSchemaForGetFavorites() throws Exception
...
 
         adminFavorites.getEntries().stream()
                 .map(RestPersonFavoritesModel::onModel)
                 .forEach(m -> m.assertThat().field(ALLOWABLE_OPERATIONS).isNotEmpty());
     }
+
+    @TestRail(section = {TestGroup.REST_API, TestGroup.FAVORITES}, executionType = ExecutionType.REGRESSION,
+            description = "Verify the get favorites request with properties parameter applied")
+    @Test(groups = {TestGroup.REST_API, TestGroup.FAVORITES, TestGroup.REGRESSION})
+    public void checkSearchResponseContainsIsFavoriteWhenRequested() throws Exception
```
Comment (tpage-alfresco) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java:579]: From PMD:
> +++ 1 new SignatureDeclareThrowsException violation(s) +++
> 
> packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java:579 A method/constructor should not explicitly throw java.lang.Exception

Does one of the methods in this throw an exception? TBH this rule doesn't seem particularly useful for test methods.

Comment (tpage-alfresco) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java:579]: Oh - it turns out we already disabled it for tests, but according to the docs this only covers JUnit 3. https://docs.pmd-code.org/latest/pmd_rules_java_design.html#signaturedeclarethrowsexception

Comment (mpichura) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java]: Nevertheless, I refactored the class and removed most of `throws Exception` declarations as the were obsolete.

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java (hunk 2 above)
Comment (mpichura) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java]: Should be okay now

**Review by tpage-alfresco — APPROVED** [review]
Nice!  PMD says you've reduced it from 40 issues to 6 issues. :+1:

**Pull request merged**